distage-example-monofunctor-tf/src/main/scala/leaderboard/LeaderboardRole.scala

```diff
@@ -0,0 +1,319 @@
+package leaderboard
+
+import cats.Applicative
+import distage.StandardAxis.Repo
+import distage.plugins.PluginConfig
+import distage.{Activation, Lifecycle, Module, ModuleDef}
+import izumi.distage.model.definition.StandardAxis.Scene
+import izumi.distage.roles.RoleAppMain
+import izumi.distage.roles.bundled.{ConfigWriter, Help}
+import izumi.distage.roles.model.{RoleDescriptor, RoleService}
+import izumi.fundamentals.platform.IzPlatform
+import izumi.fundamentals.platform.cli.model.raw.{RawEntrypointParams, RawRoleParams, RawValue}
+import logstage.LogIO
+import leaderboard.api.{LadderApi, ProfileApi}
+import leaderboard.http.HttpServer
+import leaderboard.plugins.{LeaderboardPlugin, PostgresDockerPlugin}
+import zio.interop.catz.asyncInstance
+import zio.Task
+
+import scala.annotation.unused
+
+/**
+  * A role that exposes just the /ladder/ endpoints, it can be launched with
+  *
+  * {{{
+  *   ./launcher :ladder
+  * }}}
+  *
+  * Example session:
+  *
+  * {{{
+  *   curl -X POST http://localhost:8080/ladder/50753a00-5e2e-4a2f-94b0-e6721b0a3cc4/100
+  *   curl -X GET http://localhost:8080/ladder
+  * }}}
+  */
+final class LadderRole[F[+_]: Applicative](
+  @unused ladderApi: LadderApi[F],
+  @unused runningServer: HttpServer,
+  log: LogIO[F],
+) extends RoleService[F[_]] {
+  override def start(roleParameters: RawEntrypointParams, freeArgs: Vector[String]): Lifecycle[F[_], Unit] = {
+    Lifecycle.liftF(log.info("Ladder API started!"))
+  }
+}
+object LadderRole extends RoleDescriptor {
+  final val id = "ladder"
+}
+
+/**
+  * A role that exposes just the /profile/ endpoints, it can be launched with
+  *
+  * {{{
+  *   ./launcher :profile
+  * }}}
+  *
+  * Example session:
+  *
+  * {{{
+  *   curl -X POST http://localhost:8080/profile/50753a00-5e2e-4a2f-94b0-e6721b0a3cc4 -d '{"name": "Kai", "description": "S C A L A"}'
+  *   curl -X GET http://localhost:8080/profile/50753a00-5e2e-4a2f-94b0-e6721b0a3cc4
+  * }}}
+  */
+final class ProfileRole[F[+_]: Applicative](
+  @unused profileApi: ProfileApi[F],
+  @unused runningServer: HttpServer,
+  log: LogIO[F],
+) extends RoleService[F[_]] {
```

Comment:
Please remove redundant `[_]`'s in code.
```suggestion
) extends RoleService[F] {
```